bot/cogs/snakes.py

```diff
@@ -0,0 +1,1231 @@
+import asyncio
+import colorsys
+import logging
+import os
+import random
+import re
+import string
+import textwrap
+import urllib
+from functools import partial
+from io import BytesIO
+from typing import Any, Dict
+
+import aiohttp
+import async_timeout
+from discord import Colour, Embed, File, Member, Message, Reaction
+from discord.ext.commands import AutoShardedBot, BadArgument, Context, bot_has_permissions, command
+from PIL import Image, ImageDraw, ImageFont
+
+from bot.constants import ERROR_REPLIES, OMDB_API_KEY, SITE_API_KEY, SITE_API_URL, YOUTUBE_API_KEY
+from bot.converters import Snake
+from bot.decorators import locked
+from bot.utils.snakes import hatching, perlin, perlinsneks, sal
+
+log = logging.getLogger(__name__)
+
+# region: Constants
+# Color
+SNAKE_COLOR = 0x399600
+
+# Antidote constants
+SYRINGE_EMOJI = "\U0001F489"  # :syringe:
+PILL_EMOJI = "\U0001F48A"     # :pill:
+HOURGLASS_EMOJI = "\u231B"    # :hourglass:
+CROSSBONES_EMOJI = "\u2620"   # :skull_crossbones:
+ALEMBIC_EMOJI = "\u2697"      # :alembic:
+TICK_EMOJI = "\u2705"         # :white_check_mark: - Correct peg, correct hole
+CROSS_EMOJI = "\u274C"        # :x: - Wrong peg, wrong hole
+BLANK_EMOJI = "\u26AA"        # :white_circle: - Correct peg, wrong hole
+HOLE_EMOJI = "\u2B1C"         # :white_square: - Used in guesses
+EMPTY_UNICODE = "\u200b"      # literally just an empty space
+
+ANTIDOTE_EMOJI = [
+    SYRINGE_EMOJI,
+    PILL_EMOJI,
+    HOURGLASS_EMOJI,
+    CROSSBONES_EMOJI,
+    ALEMBIC_EMOJI,
+]
+
+# Quiz constants
+ANSWERS_EMOJI = {
+    "a": "\U0001F1E6",  # :regional_indicator_a: 🇦
+    "b": "\U0001F1E7",  # :regional_indicator_b: 🇧
+    "c": "\U0001F1E8",  # :regional_indicator_c: 🇨
+    "d": "\U0001F1E9",  # :regional_indicator_d: 🇩
+}
+
+ANSWERS_EMOJI_REVERSE = {
+    "\U0001F1E6": "A",  # :regional_indicator_a: 🇦
+    "\U0001F1E7": "B",  # :regional_indicator_b: 🇧
+    "\U0001F1E8": "C",  # :regional_indicator_c: 🇨
+    "\U0001F1E9": "D",  # :regional_indicator_d: 🇩
+}
+
+# Zzzen of pythhhon constant
+ZEN = """
```

Comment:
```py
import this

ZEN = ''.join(this.d.get(char) or char for char in this.s)
```
?

Reply:
actually, i forgot that spams console
```py
import contextlib
import os


with contextlib.redirect_stdout(os.devnull):
    import this
ZEN = ''.join(this.d.get(char) or char for char in this.s)
```

Reply:
naw. it's a fun way to do it, but infinitely less readable.

Reply:
:(